src/mysqlsh_helpers.py

```diff
@@ -35,84 +66,172 @@ class MySQL:
 
     def __init__(
         self,
+        instance_address: str,
+        cluster_name: str,
         root_password: str,
+        server_config_user: str,
+        server_config_password: str,
         cluster_admin_user: str,
         cluster_admin_password: str,
-        instance_address: str,
     ):
         """Initialize the MySQL class.
 
         Args:
+            instance_address: address of the targeted instance
+            cluster_name: cluster name
             root_password: password for the 'root' user
+            server_config_user: user name for the server config user
+            server_config_password: password for the server config user
             cluster_admin_user: user name for the cluster admin user
             cluster_admin_password: password for the cluster admin user
-            instance_address: address of the targeted instance
         """
+        self.instance_address = instance_address
+        self.cluster_name = cluster_name
         self.root_password = root_password
+        self.server_config_user = server_config_user
+        self.server_config_password = server_config_password
         self.cluster_admin_user = cluster_admin_user
         self.cluster_admin_password = cluster_admin_password
-        self.instance_address = instance_address
 
-    @property
-    def mysqlsh_bin(self) -> str:
+    @staticmethod
+    def _get_mysqlsh_bin() -> str:
```

Comment:
Again I could be wrong, but the pattern of underscore-prefixing object methods is usually to indicate that this method shouldn't be used from outside the object. 

Whereas a `staticmethod` tells me that it's also able to be used from outside the object.

Reply:
You are right! Somehow, my reasoning was that it's a staticmethod only used within the context of the object (and so it's private). But there is no such thing!

Updated to public in [7dbb768](https://github.com/canonical/mysql-operator/pull/6/commits/7dbb7684b4aab2198d0bdae74307b83f093fbb8d)